CPU_Simulator/src/main/java/unit/Control.java

```diff
@@ -0,0 +1,111 @@
+package unit;
+
+import lombok.Data;
+import register.*;
+
+import java.util.*;
+
+@Data
+public class Control implements Unit {
+    private Map<String, Unit> units;
+    private Map<String, Register> registers;
+    private String instruction;
+    private String aluOperation;
+    private boolean registerDestination;
+    private boolean extendedOperation;
+    private boolean branch;
+    private boolean jump;
+    private boolean aluSource;
+    private boolean memoryWrite;
+    private boolean memoryToRegister;
+    private boolean registerWrite;
+
+    public Control() {
+        units = new HashMap<>();
+        registers = new HashMap<>();
+        instruction = "000";
+        aluOperation = "000";
+    }
+
+    @Override
+    public void run() {
+        aluOperation = "000";
+        registerDestination = false;
+        extendedOperation = false;
+        branch = false;
+        jump = false;
+        memoryWrite = false;
+        memoryToRegister = false;
+        registerWrite = false;
+        aluSource = false;
+
+        switch (instruction) {
+            case "000": //R
```

Comment:
I think it would be more readable if you would define constants for these as well